src/main/java/org/jitsi/videobridge/Content.java

```diff
@@ -259,20 +259,28 @@ void askForKeyframes(Collection<Endpoint> endpoints)
      * @param initiator the value to use for the initiator field, or
      * <tt>null</tt> to use the default value.
      * @param rtpLevelRelayType
+     * @param endpointId the id of the endpoint this channel is for, if provided
+     * the endpoint id will be used as the channel id
      * @return the created <tt>RtpChannel</tt> instance.
      * @throws Exception
      */
     public RtpChannel createRtpChannel(String channelBundleId,
                                        String transportNamespace,
                                        Boolean initiator,
-                                       RTPLevelRelayType rtpLevelRelayType)
+                                       RTPLevelRelayType rtpLevelRelayType,
+                                       String endpointId)
         throws Exception
     {
         RtpChannel channel = null;
 
         do
         {
-            String id = generateChannelID();
+            String id;
+            if(!StringUtils.isNullOrEmpty(endpointId)) {
+                id = endpointId;
+            } else {
+                id = generateChannelID();
+            }
```

Comment:
If it so happens that there is already a channel with the ID of the endpoint, this will enter an infinite loop. Can we throw an exception instead?
Also, can you please place the braces on separate lines, as per our convention?